interface/H4LTools.h

```diff
@@ -0,0 +1,176 @@
+
+      bool ZZSelection();
+      TLorentzVector Z1;
+      TLorentzVector Z2;
+
+      RoccoR  *calibrator;
+      //Mela* mela;
+      float ApplyRoccoR(bool isMC, int charge, float pt, float eta, float phi, float genPt, float nLayers);
+      std::vector<float> Muon_Pt_Corrected;
+      void MuonPtCorrection(bool isMC);
+
+    private:
+
+      TTreeReaderValue<unsigned> *nElectron = nullptr;
+      TTreeReaderArray<float> *Electron_pt = nullptr;
+      TTreeReaderArray<float> *Electron_eta = nullptr;
+      TTreeReaderArray<float> *Electron_phi = nullptr;
+      TTreeReaderArray<float> *Electron_mass = nullptr;
+      TTreeReaderArray<float> *Electron_dxy = nullptr;
+      TTreeReaderArray<float> *Electron_dz = nullptr;
+      TTreeReaderArray<float> *Electron_sip3d = nullptr;
+      TTreeReaderArray<bool> *Electron_mvaFall17V2Iso_WP90 = nullptr;
+      TTreeReaderArray<int> *Electron_pdgId = nullptr;
+
+      TTreeReaderValue<unsigned> *nMuon = nullptr;
+      TTreeReaderArray<float> *Muon_pt = nullptr;
+      TTreeReaderArray<float> *Muon_eta = nullptr;
+      TTreeReaderArray<float> *Muon_phi = nullptr;
+      TTreeReaderArray<float> *Muon_mass = nullptr;
+      TTreeReaderArray<bool> *Muon_isGlobal = nullptr;
+      TTreeReaderArray<bool> *Muon_isTracker = nullptr;
+      TTreeReaderArray<bool> *Muon_isPFcand = nullptr;
+      TTreeReaderArray<int> *Muon_nTrackerLayers = nullptr;
+      TTreeReaderArray<float> *Muon_sip3d = nullptr;
+      TTreeReaderArray<float> *Muon_dxy = nullptr;
+      TTreeReaderArray<float> *Muon_dz = nullptr;
+      TTreeReaderArray<float> *Muon_ptErr = nullptr;
+      TTreeReaderArray<int> *Muon_pdgId = nullptr;
+      TTreeReaderArray<int> *Muon_charge = nullptr;
+      TTreeReaderArray<float> *Muon_pfRelIso03_all = nullptr;
+      TTreeReaderArray<int> *Muon_genPartIdx = nullptr;
+
+      TTreeReaderValue<unsigned> *nFsrPhoton = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_dROverEt2 = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_eta = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_phi = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_pt = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_relIso03 = nullptr;
+      TTreeReaderArray<float> *FsrPhoton_muonIdx = nullptr;
+
+      TTreeReaderValue<unsigned> *nGenPart = nullptr;
+      TTreeReaderArray<float> *GenPart_pt = nullptr;
+
+      TTreeReaderValue<unsigned> *nJet = nullptr;
+      TTreeReaderArray<float> *Jet_eta = nullptr;
+      TTreeReaderArray<float> *Jet_phi = nullptr;
+      TTreeReaderArray<float> *Jet_pt = nullptr;
+      TTreeReaderArray<float> *Jet_mass = nullptr;
+      TTreeReaderArray<float> *Jet_btagDeepC = nullptr;
+      TTreeReaderArray<int> *Jet_jetId = nullptr;
+      TTreeReaderArray<int> *Jet_puId = nullptr;
+
+
+
+
+
+      
+
+      
+
+};
+
+H4LTools::H4LTools(){
+  std::string DATAPATH = "/afs/cern.ch/user/y/yujil/Run3H4l/CMSSW_10_6_20/src/PhysicsTools/NanoAODTools/python/postprocessing/analysis/nanoAOD_vvVBS";
```

Comment:
Put the relative path instead of the absolute path.

This is pointing to your afs path, and it won't work for others.

Reply:
One of easiest way that I can think of is:

Just replace:

```cpp
  std::string DATAPATH = "/afs/cern.ch/user/y/yujil/Run3H4l/CMSSW_10_6_20/src/PhysicsTools/NanoAODTools/python/postprocessing/analysis/nanoAOD_vvVBS";
```

with 

```cpp
  std::string DATAPATH = "";
```

and run the code from path:

```bash
/afs/cern.ch/user/y/yujil/Run3H4l/CMSSW_10_6_20/src/PhysicsTools/NanoAODTools/python/postprocessing/analysis/nanoAOD_vvVBS
```